specification/storage/data-plane/Microsoft.StorageDataLake/stable/2019-12-12/DataLakeStorage.json

```diff
@@ -2189,7 +2189,29 @@
                 "type": "string"
               },
               "x-ms-continuation": {
-                "description": "When deleting a directory, the number of paths that are deleted with each invocation is limited.  If the number of paths to be deleted exceeds this limit, a continuation token is returned in this response header.  When a continuation token is returned in the response, it must be specified in a subsequent invocation of the delete operation to continue deleting the directory.",
+                "description": "When deleting a directory, the number of paths that are deleted with each invocation is limited.  If the number of paths to be deleted exceeds this limit, a continuation token is returned in this response header for non-hierarchical namespace accounts. But for hierarchical namespace enabled accounts, a continuation token is returned only when the following conditions are met: Delete should be on a directory with query parameters \"recursive\" and \"paginated\" set to true, the identity of the user calling the api should be a non-super user using ACL based authorization, number of paths to be deleted exceeds a limit, request version is 2023-08-03 and later. When a continuation token is returned in the response, it must be specified in a subsequent invocation of the delete operation to continue deleting the directory. Directory is successfully deleted when either the continuation token returned is empty in case of hierarchical namespace accounts and when it is not present in the response otherwise. The actual directory deletion for hierarchical namespace enabled accounts happens only in the last invocation, the previous ones involve ACL checks in the server of the files and directories under the directory to be recursively deleted.",
```

Comment:
_Directory is successfully deleted when either the continuation token returned is empty in case of hierarchical namespace accounts and when it is not present in the response otherwise._
Does this imply that for non-HNS enabled accounts, successful directory deletion doesn't return this header in response? Should we change ` and when it is not present in the response otherwise.` to `or when it is not present in the response.`?

_the previous ones involve ACL checks in the server of the files and directories under the directory to be recursively deleted._

How about `the previous delete calls involve ACL checks on the files and directories under the directory to be deleted recursively.`?

Reply:
Logical - made the proposed changes